src/radical/entk/execman/rp/task_processor.py

```diff
@@ -170,12 +180,11 @@ def resolve_tags(tag, parent_pipeline_name, placeholders):
 
         for sname in placeholders[pname]:
             for tname in placeholders[pname][sname]:
-                if tag != tname:
+                if colo_tag != tname:
                     continue
-                return placeholders[pname][sname][tname]['rts_uid']
+                return placeholders[pname][sname][tname]['uid']
```

Comment:
Yes, there can be only one match. Let's say that in every pipeline there is a tag to the same pipeline. Placing everything in a loop will be O(n**2), where n is the number of pipelines. How it is right now, it will use the loop if and only if the tag is from a different pipeline. I'm perfectly fine changing it, but I do not see any benefits, apart from simpler code.

Reply:
Yeah, I like simpler code :-)  But if that assumption holds (dependencies are mostly to the parent pipeline), I see the point.